pattern/pattern_test.go

```diff
@@ -18,65 +18,65 @@ var translateTests = []struct {
 	{pat: `foóà中`, mode: Filenames | Braces, want: `foóà中`},
-	{pat: `.`, want: `\.`},
-	{pat: `foo*`, want: `foo.*`},
-	{pat: `foo*`, mode: Shortest, want: `foo.*?`},
-	{pat: `foo*`, mode: Shortest | Filenames, want: `foo[^/]*?`},
-	{pat: `*foo`, mode: Filenames, want: `[^/]*foo`},
-	{pat: `**`, want: `.*.*`},
-	{pat: `**`, mode: Filenames, want: `.*`},
-	{pat: `/**/foo`, want: `/.*.*/foo`},
-	{pat: `/**/foo`, mode: Filenames, want: `/(.*/|)foo`},
-	{pat: `/**/à`, mode: Filenames, want: `/(.*/|)à`},
-	{pat: `/**foo`, mode: Filenames, want: `/.*foo`},
-	{pat: `\*`, want: `\*`},
+	{pat: `.`, want: `(?s)\.`},
+	{pat: `foo*`, want: `(?s)foo.*`},
+	{pat: `foo*`, mode: Shortest, want: `(?s)foo.*?`},
+	{pat: `foo*`, mode: Shortest | Filenames, want: `(?s)foo[^/]*?`},
+	{pat: `*foo`, mode: Filenames, want: `(?s)[^/]*foo`},
+	{pat: `**`, want: `(?s).*.*`},
+	{pat: `**`, mode: Filenames, want: `(?s).*`},
+	{pat: `/**/foo`, want: `(?s)/.*.*/foo`},
+	{pat: `/**/foo`, mode: Filenames, want: `(?s)/(.*/|)foo`},
+	{pat: `/**/à`, mode: Filenames, want: `(?s)/(.*/|)à`},
+	{pat: `/**foo`, mode: Filenames, want: `(?s)/.*foo`},
+	{pat: `\*`, want: `(?s)\*`},
 	{pat: `\`, wantErr: true},
-	{pat: `?`, want: `.`},
-	{pat: `?`, mode: Filenames, want: `[^/]`},
-	{pat: `?à`, want: `.à`},
-	{pat: `\a`, want: `a`},
-	{pat: `(`, want: `\(`},
-	{pat: `a|b`, want: `a\|b`},
-	{pat: `x{3}`, want: `x\{3\}`},
-	{pat: `{3,4}`, want: `\{3,4\}`},
-	{pat: `{3,4}`, mode: Braces, want: `(?:3|4)`},
-	{pat: `{3,`, want: `\{3,`},
-	{pat: `{3,`, mode: Braces, want: `\{3,`},
-	{pat: `{3,{4}`, mode: Braces, want: `\{3,\{4\}`},
-	{pat: `{3,{4}}`, mode: Braces, want: `(?:3|\{4\})`},
-	{pat: `{3,{4,[56]}}`, mode: Braces, want: `(?:3|(?:4|[56]))`},
-	{pat: `{3..5}`, mode: Braces, want: `(?:3|4|5)`},
-	{pat: `{9..12}`, mode: Braces, want: `(?:9|10|11|12)`},
-	{pat: `[a]`, want: `[a]`},
-	{pat: `[abc]`, want: `[abc]`},
-	{pat: `[^bc]`, want: `[^bc]`},
-	{pat: `[!bc]`, want: `[^bc]`},
-	{pat: `[[]`, want: `[[]`},
-	{pat: `[\]]`, want: `[\]]`},
-	{pat: `[\]]`, mode: Filenames, want: `[\]]`},
-	{pat: `[]]`, want: `[]]`},
-	{pat: `[!]]`, want: `[^]]`},
-	{pat: `[^]]`, want: `[^]]`},
-	{pat: `[a/b]`, want: `[a/b]`},
-	{pat: `[a/b]`, mode: Filenames, want: `\[a/b\]`},
+	{pat: `?`, want: `(?s).`},
+	{pat: `?`, mode: Filenames, want: `(?s)[^/]`},
+	{pat: `?à`, want: `(?s).à`},
+	{pat: `\a`, want: `(?s)a`},
+	{pat: `(`, want: `(?s)\(`},
+	{pat: `a|b`, want: `(?s)a\|b`},
+	{pat: `x{3}`, want: `(?s)x\{3\}`},
+	{pat: `{3,4}`, want: `(?s)\{3,4\}`},
+	{pat: `{3,4}`, mode: Braces, want: `(?s)(?:3|4)`},
+	{pat: `{3,`, want: `(?s)\{3,`},
+	{pat: `{3,`, mode: Braces, want: `(?s)\{3,`},
+	{pat: `{3,{4}`, mode: Braces, want: `(?s)\{3,\{4\}`},
+	{pat: `{3,{4}}`, mode: Braces, want: `(?s)(?:3|\{4\})`},
+	{pat: `{3,{4,[56]}}`, mode: Braces, want: `(?s)(?:3|(?:4|[56]))`},
+	{pat: `{3..5}`, mode: Braces, want: `(?s)(?:3|4|5)`},
+	{pat: `{9..12}`, mode: Braces, want: `(?s)(?:9|10|11|12)`},
+	{pat: `[a]`, want: `(?s)[a]`},
```

Comment:
nit: patterns like these don't need `(?s)` because they do not use `.`. I understand that it probably doesn't make a difference in terms of performance, but as an API, it would be nice if `pattern.Regexp` doesn't return extra characters unless it really needs to.

One option is to always write `(?s)` into the buffer, and slice the prefix out of the returned string if we didn't write `.` nor `.*`.

Then the tests won't change as much either :)

Reply:
Thanks for pointing this out, I wasn't in love with my approach either. I have taken your suggestion which definitely cleans up the diff. Hopefully, this won't cause too much consternation when I get back to working on my patch for nocaseglob 😉.